src/site_upload/update_metadata/update_metadata.py

```diff
@@ -0,0 +1,63 @@
+"""Lambda for updating metadata
+
+Note: since this lambda is used with an EventSource of a FIFO queue,
+this lambda will not be allowed to be concurrently executed. Instead,
+it will be executed once, and fed up to ten events from that queue
+to be run in batch. Once it terminates, it will be re-invoked if there
+are remaining events in the queue.
+"""
+
+import json
+import os
+
+import boto3
+import botocore
+
+from shared import decorators, functions
+
+s3_client = boto3.client("s3")
+
+
+def update_source(metadata, update):
+    for update_key, update_val in update.items():
+        if isinstance(update_val, dict):
+            metadata[update_key] = update_source(
+                metadata[update_key] if update_key in metadata.keys() else {}, update[update_key]
+            )
+        else:
+            if not (
```

Comment:
nit 1: `elif not ...`

nit 2: I would remove the `not` and flip the conditional logic:
```
  update_val is not None
  or update_key not in metadata
  or metadata[update_key] is None
```
It's easier to reason about for my brain, anyway.

nit 3: you could reduce the clauses by one if you did:
```
  update_val is not None
  or metadata.get(update_key) is None
```